name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/columns/DividendPaymentColumn.java

```diff
@@ -0,0 +1,170 @@
+import name.abuchen.portfolio.ui.util.Colors;
+import name.abuchen.portfolio.ui.util.viewers.Column;
+import name.abuchen.portfolio.ui.util.viewers.ColumnViewerSorter;
+import name.abuchen.portfolio.ui.util.viewers.DateLabelProvider;
+
+public class DividendPaymentColumn
+{
+
+    static Column createNextDividendExDateColumn()
+    {
+        Column column = new Column("nexdd", Messages.ColumnDividendsNextExDate, SWT.LEFT, 80); //$NON-NLS-1$
+        column.setMenuLabel(Messages.ColumnDividendsNextExDate_MenuLabel);
+        column.setGroupLabel(Messages.GroupLabelDividends);
+        column.setVisible(false);
+
+        LocalDate now = LocalDate.now();
+
+        Function<Object, LocalDate> dataProvider = element -> {
+            Security sec = Adaptor.adapt(Security.class, element);
+            if (sec == null)
+            {
+                return null; //
+            }
+            return sec.getEvents().stream() //
+                            .filter(DividendEvent.class::isInstance) //
+                            .map(e -> ((DividendEvent) e).getDate()) //
+                            .sorted(Comparable::compareTo) //
+                            .filter(d -> d.isAfter(now)) //
+                            .findFirst().orElse(null);
+        };
+
+        column.setLabelProvider(new DateLabelProvider(dataProvider)
+        {
+            @Override
+            public Color getBackground(Object element)
+            {
+                LocalDate date = dataProvider.apply(element);
+                if (date == null)
+                {
+                    return null; //
+                }
+                if (!date.equals(LocalDate.now()))
+                {
+                    return null; //
+                }
+                return Colors.theme().greenBackground();
+            }
+        });
+        column.setSorter(ColumnViewerSorter.create(dataProvider::apply));
+        return column;
+    }
+
+    static Column createNextDividendPaymentDateColumn()
+    {
+        Column column = new Column("nexpdd", Messages.ColumnDividendsNextPaymentDate, SWT.LEFT, 80); //$NON-NLS-1$
+        column.setMenuLabel(Messages.ColumnDividendsNextPaymentDate_MenuLabel);
+        column.setGroupLabel(Messages.GroupLabelDividends);
+        column.setVisible(false);
+
+        LocalDate now = LocalDate.now();
+
+        Function<Object, LocalDate> dataProvider = element -> {
+            Security sec = Adaptor.adapt(Security.class, element);
+            if (sec == null)
+            {
+                return null; //
+            }
+            return sec.getEvents().stream() //
+                            .filter(DividendEvent.class::isInstance) //
+                            .map(e -> ((DividendEvent) e).getPaymentDate()) //
+                            .sorted(Comparable::compareTo) //
+                            .filter(d -> d.isAfter(now)) //
+                            .findFirst().orElse(null);
+        };
```

Comment:
```suggestion
LocalDate now = LocalDate.now();

Function<Object, LocalDate> dataProvider = element -> {
    Security security = Adaptor.adapt(Security.class, element);
    return Optional.ofNullable(security)
                   .map(s -> s.getEvents().stream()
                               .filter(DividendEvent.class::isInstance)
                               .map(e -> ((DividendEvent) e).getPaymentDate())
                               .filter(d -> d.isAfter(now))
                               .min(Comparable::compareTo)
                               .orElse(null))
                   .orElse(null);
};
```

Reply:
Same answer as in https://github.com/portfolio-performance/portfolio/pull/3927#discussion_r1560601973